src/plugins/robot-memory/robot_memory.cpp

```diff
@@ -877,9 +877,32 @@ RobotMemory::mutex_try_lock(const std::string& name,
 		        new_doc.getField("locked").Bool());
 
 	} catch (mongo::OperationException &e) {
-		//if (e.obj()["code"].numberInt() != 11000) {
```

Comment:
As the comment indicates, some error is expected here. Does the observed behavior depend on some specific error code? If so, it may be worth omitting the check if not required. But this is minor and would not block the review to go through.

Reply:
Good question, I never checked the error code. I think we always saw some generic database error and it wasn't clear what exactly went wrong.

Reply:
Alright then. Let's take the performance overhead here for robustness.